OneCommunityGlobal/HighestGoodNetworkApp
# Nahiyan-Project's tab overhaul

# Description
Issues:

1. When deleting a project from the projects tab within the user profile, the user can assign another project without saving it. This is incorrect because when the user attempts to add another project, it will not work. 
2. No tooltip on hover over the 'Assign Project' button when disabled.
3. The Cancel button would not revert the changes made to projects
4. Users who aren't able to delete projects or tasks can see the delete button column
5. When deleting a project, that project row disappears but the tasks affiliated with it do not. (Only after saving do they disappear) 
6. Buttons are not on the same line as the tasks header or the projects header
7. When clicking "Assign Project", the input should be autofocus on the popup

## Desktop Version:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/c7283701-6281-45d4-948f-e559e0462321
## Mobile Version:
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/2eb14ef4-5fb1-472f-a1f0-c62d5d29f4f1

## Related PRS (if any):
Use development backend

## Main changes explained:
- Cursor automatically on popup input after clicking on the `Assign project` button
- The `Assign project` button is disabled when deleting a project
- Also the same for tablet/phone view
- When deleting a project, it also deletes the affiliated tasks.
- Users who are unable to delete projects, cannot see the delete buttons column
- Users who are unable to update tasks(add/remove) cannot see the add/remove columns
- Clicking on cancel will now revert any changes made to the tasks or projects
- Clicking on the 'X' button on mobile view will now revert any changes made to tasks or projects
- Extracted styling to CSS file or incorporated within the class name
- Moved the `Assignn Button`  to be on the same line as the Projects header
- Moved the tasks filter buttons to be on the same line as the Tasks header

## How to test:
1. check into the current branch
8. do `npm install` and `npm run start:local` to run this PR locally
9. Clear site data/cache
10. log in as owner user
11. Dashboard -> User Profile -> Projects Tab
12. Make sure the cursor is automatically on the popup input when clicking on `Assign project`
13. Add any project 
14. Delete any project
15. The following should occur: 
- `Assign project` button is disabled
- When the `save changes` button is clicked, then the `Assign project` button will be enabled again
16. Steps 5-6 should be repeated in a tablet or phone view (width lower than 1025px)
17. Make sure the `Assign project` button is disabled after deleting a project
18. Assign tasks to this user, check out this video on how to do that, [Video](https://www.loom.com/share/8bc672c323b64955aac4bce46c3794aa?sid=92b7b345-237d-4c9f-b05d-d16aebacf90f)
19. Attempt to delete a project (Do not hit save changes), the tasks affiliated with that should disappear
20. Click on cancel. 
21. Project(s) and tasks should have reverted to original

## Changes
https://github.com/OneCommunityGlobal/HighestGoodNetworkApp/assets/71025942/ea80f395-504e-49ee-b659-eb2eda8475fd


## If you have any issues testing, please let me know on Slack
## Also please test the functionality in both mobile view and desktop
